src/app/components/dynamicdialog/dynamicdialog.ts

```diff
@@ -18,44 +18,113 @@ const hideAnimation = animation([
 ]);
 
 @Component({
-	selector: 'p-dynamicDialog',
-	template: `
-        <div #mask [ngClass]="{'p-dialog-mask':true, 
-            'p-component-overlay p-component-overlay-enter p-dialog-mask-scrollblocker': config.modal !== false, 
-            'p-dialog-left': position === 'left', 
+    selector: 'p-dialog-title',
+    template:`<div class="p-dialog-header" (mousedown)="initDrag($event)">
+    <span class="p-dialog-title">{{config.header}}</span>
+    <div class="p-dialog-header-icons">
+      <button *ngIf="config.maximizable" type="button" [ngClass]="{'p-dialog-header-icon p-dialog-header-maximize p-link':true}" (click)="maximize()" (keydown.enter)="maximize()" tabindex="-1" pRipple>
+        <span class="p-dialog-header-maximize-icon" [ngClass]="maximized ? minimizeIcon : maximizeIcon"></span>
+      </button>
+      <button [ngClass]="'p-dialog-header-icon p-dialog-header-maximize p-link'" type="button" (click)="hide()" (keydown.enter)="hide()" *ngIf="config.closable !== false">
+        <span class="p-dialog-header-close-icon pi pi-times"></span>
+      </button>
+    </div>
+  </div>`
+})
+export class DynamicDialogTitle {
+
+    get config(): DynamicDialogConfig {
+        return this.parent.config;
+    }
+
+    get maximizable(): boolean {
+        return this.config.maximizable;
+    }
+
+    get maximizeIcon(): string {
+        return this.config.maximizeIcon ? this.config.maximizeIcon : 'pi pi-window-maximize';
+    }
+
+    get minimizeIcon(): string {
+        return this.config.minimizeIcon ? this.config.minimizeIcon : 'pi pi-window-minimize';
+    }
+
+    get maximized() {
+        return this.parent.maximized
+    }
+
+    set maximized(value) {
+        this.parent.maximized = value
+    }
+
+    maximize() {
+        this.maximized = !this.maximized;
+
+        if(this.maximized) {
+            DomHandler.addClass(document.body, 'p-overflow-hidden');
+        }
+        else {
+            DomHandler.removeClass(document.body, 'p-overflow-hidden');
+        }
+
+        this.parent.dialogRef.maximize({'maximized': this.maximized})
+    }
+
+    hide() { this.parent.hide() }
+
+    initDrag($event) { this.parent.initDrag($event) }
+
+    constructor (public parent: DynamicDialogComponent) {
+        console.log('title component')
```

Comment:
I think this should be deleted.

Reply:
Oversight when copying code to the PR branch, removed now.